backend/src/services/secret-sharing/secret-sharing-service.ts

```diff
@@ -105,9 +112,21 @@ export const secretSharingServiceFactory = ({
     return userSharedSecrets;
   };
 
-  const getActiveSharedSecretByIdAndHashedHex = async (sharedSecretId: string, hashedHex: string) => {
+  const getActiveSharedSecretByIdAndHashedHex = async (sharedSecretId: string, hashedHex: string, orgId?: string) => {
     const sharedSecret = await secretSharingDAL.findOne({ id: sharedSecretId, hashedHex });
     if (!sharedSecret) return;
+
+    const orgName = sharedSecret.orgId ? (await orgDAL.findOrgById(sharedSecret.orgId))?.name : "";
+    // Support organization level access for secret sharing
+    if (sharedSecret.accessType === SecretSharingAccessType.Organization && orgId !== sharedSecret.orgId) {
```

Comment:
I think it might make more sense to throw an error here because the user is trying to view a secret share for a different org than they are logged in as?

Reply:
But it's just validation, right?
If an error is returned, we'll lost the orgName in the frontend.
The idea of this logic is expose the organization name to bring context to the user, so they know that they need to log in into XY organization to get access.